guide/jupyter_guide.adoc

```diff
@@ -0,0 +1,225 @@
+
+```python
+ex = ExecutePreprocessor(kernel_name=<kernel>)
+ex.km, ex.kc = ex.start_new_kernel()
+```
+
+For the value of *kernel_name* set it to python2 or python3, depending on what version used to make the Jupyter Notebook. The python kernel is already required for any notebook, regardless of kernel, so nothing else extra is required.
+
+[NOTE]
+====
+Other non-python kernels may be supported in the future. For those, you would need to install them in your docker file like *IPython*.
+====
+
+====== Execution
+Execution is done on a cell-by-cell basis. This is done primarily to get better error checking out of the notebook. If the notebook is executed all at once, the only exception that can be caught is the *CellExecutionError* with a return code of 1. By running the cells sequentially, it is possible to get return codes based on individual cells as opposed to the notebook as a whole. Doing this is as simple as a try-catch and a for loop.
+
+```python
+return_code = 0
+cell_count = 0
+try:
+for cell in nb.cells:
+    ex.preprocess_cell(cell, None, cell_count)  
+    cell_count += 1
+except CellExecutionError as error: 
+    error_string = str(error)
+    ansi_escape = re.compile(r'(\x9B|\x1B\[)[0-?]*[ -\/]*[@-~]')
+    cleaned_error_string = ansi_escape.sub('', error_string)
+    print(cleaned_error_string, file=sys.stderr)
+
+    #<generate-return-code> 
+finally:
+    ex.km.shutdown_kernel() 
+```
+
+Generation of the return code can be done a variety of ways. This is up to the discretion of the developer. One way is to look at the cell's metadata and base it off the json data stored within it. An example of this can be seen below.
+
+```python
+metadata = cell.metadata
+if "code" in metadata:
+    return_code = metadata["code"]
+else:
+    return_code = ERROR_GENERIC_CELL
+```
+
+====== Write to Notebook
+Writing to a notebook is very similar in implementation to reading a notebook. As before, a output notebook must first be opened (and created) and then the *nbformat* tool is used to write the executed cells to the output notebook.
+
+```python
+f = open(nbFileName, "w")
+nbformat.write(nb, f)
+f.close()
+```
+
+===== Running in Docker
+First, create a script called image_rotate.sh in the same directory as the Python file created above:
+
+```sh
+include::example/jupyter/image_rotate.sh[]
+```
+
+We will use the script  to invoke out Jupyter algorithm
+
+Create a file named Dockerfile in the same directory:
+
+```
+include::example/jupyter/Dockerfile[]
+```
+
+With these files, we can create our initial Docker containerized sample algorithm. Issue the following terminal commands to build the Docker image:
+
+```sh
+sudo docker build -t img-test .
```

Comment:
Let's use a different Docker image tag so to differentiate from the example in the main Seed tutorial. Something like "nb-img-test" is fine.